### PR TITLE
Add PHP Composer example

### PR DESCRIPTION
**Pull request by Crease29**

### CODE REVIEW

**Comment by ruudk** [comment]
@Crease29 Can you rebase this PR?

**Comment by Crease29** [comment]
Done. Would you mind if I re-order the examples in that file by alphabet?

**Comment by Crease29** [comment]
Hi @joshmgross, thanks for the link! I didn't know that yet!
Is the directory `~/.composer/cache/files` available in the container? If so, I'd change the path in the `examples.md` 🙂

**Comment by joshmgross** [comment]
`~/.composer/cache/files` should be available

**Comment by shivammathur** [comment]
On windows `~/.composer/cache/files` will not work as cache directory on windows will be
`C:\Users\runneradmin\AppData\Local\Composer\files`

So for a cross platform solution cache directory can be found using.
```bash
composer config cache-files-dir
```

**Comment by joshmgross** [comment]
@shivammathur Thanks!

@Crease29 

You can use that value with an action output, like:

```yaml
- name: Get Composer Cache Directory
  id: composer-cache
  run: |
    echo "::set-output name=dir::$(composer config cache-files-dir)"
- uses: actions/cache@preview
  with:
    path: ${{ steps.composer-cache.outputs.dir }}
    key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.lock') }}
    restore-keys: |
      ${{ runner.os }}-composer-
```

I've verified that the `composer config cache-files-dir` command works on all hosted runners.
- Ubuntu: `/home/runner/.composer/cache/files`
- Windows: `C:/Users/runneradmin/AppData/Local/Composer/files`
- Mac: `/Users/runner/.composer/cache/files`

**Comment by Crease29** [comment]
Thank you all for your contributions to this PR! I adjusted my changes now.

Two things that I saw in that file:
* I'd like to re-order the examples alphabetically.
* The Yarn and Nuget examples also use `~` as home directory identifier. Should that be updated to a cross-platform solution as well?

**Comment by joshmgross** [comment]
@Crease29 Thanks!

@hugovk re-ordered the examples here: https://github.com/actions/cache/pull/52 (Looks like that's caused merge conflicts for this PR)

>The Yarn and Nuget examples also use ~ as home directory identifier. Should that be updated to a cross-platform solution as well?

`~` is valid in those cases as they're still under the home directory, whereas this is under `~/AppData/Local` on Windows

File: examples.md [hunk 1]
```diff
@@ -80,12 +84,12 @@ Using [NuGet lock files](https://docs.microsoft.com/nuget/consume-packages/packa
 ## Swift, Objective-C - Carthage
 
 ```yaml
-uses: actions/cache@preview
-      with:
-        path: Carthage
-        key: ${{ runner.os }}-carthage-${{ hashFiles('**/Cartfile.resolved') }}
-        restore-keys: |
-          ${{ runner.os }}-carthage-
+- uses: actions/cache@preview
```
Comment (joshmgross) [inline examples.md]: This was fixed in https://github.com/actions/cache/pull/57, not needed for this PR

**Comment by Crease29** [comment]
Alright, I squashed my commits. Should be all fine now 🙂

File: examples.md [hunk 2]
```diff
@@ -89,6 +90,21 @@ Using [NuGet lock files](https://docs.microsoft.com/nuget/consume-packages/packa
...
 ```
 
+## PHP - Composer
+
+```yaml  
+- name: Get Composer Cache Directory
+  id: composer-cache
+  run: |
+    echo "::set-output name=dir::$(composer config cache-files-dir)"
+- uses: actions/cache@preview
```
Comment (joshmgross) [inline examples.md]: Minor, can you update this to `@v1`?

Comment (Crease29) [reply examples.md]: Done

**Pull request merged**